packages/components/src/RandomAreaPlotAnimation.tsx

```diff
@@ -5,23 +5,49 @@ import React, { useEffect, useState, useRef } from 'react';
 import debounce from 'lodash.debounce';
 import { assertNotNull } from '@deephaven/utils';
 import './RandomAreaPlotAnimation.scss';
-import ThemeExport from './ThemeExport';
+import { resolveCssVariablesInRecord, useTheme } from './theme';
 
 const VOLATILITY = 0.025; // how spikey the data gets
 const LOW = 0.9;
 const HIGH = 0.1;
 const START_Y = 0.618;
-const FOREGROUND = ThemeExport['primary-dark'];
 const GRID_SIZE = 80;
-const GRID_COLOR = ThemeExport['gray-800'];
-const BACKGROUND = ThemeExport['gray-900'];
-const PATTERN = { SIZE: 8, DOT_SIZE: 2, FILL_OPACITY: 0.25 };
+const PATTERN = { SIZE: 8, DOT_SIZE: 2 };
 const PX_PER_SECOND = 18;
 const RESIZE_DEBOUNCE = 250;
 const INTERACTION_TIMEOUT = 60 * 1000;
 
+interface ThemeColors {
+  background: string;
+  foregroundFill: string;
+  foregroundStroke: string;
+  gridColor: string;
+}
+
+const THEME_COLOR_VARIABLES = {
+  background: 'var(--dh-color-random-area-plot-animation-bg)',
+  foregroundFill: 'var(--dh-color-random-area-plot-animation-fg-fill)',
+  foregroundStroke: 'var(--dh-color-random-area-plot-animation-fg-stroke)',
+  gridColor: 'var(--dh-color-random-area-plot-animation-grid)',
+} satisfies ThemeColors;
+
+/**
+ * Resolve theme colors needed for our animation.
+ */
+function getRandomAreaPlotAnimationThemeColors(): ThemeColors {
+  return resolveCssVariablesInRecord(THEME_COLOR_VARIABLES);
+}
+
 // Draw a background canvas, paint it with a fun chart looking animation
 const RandomAreaPlotAnimation = React.memo(() => {
+  const { activeThemes } = useTheme();
+
+  const [themeColors, setThemeColors] = useState<ThemeColors | null>(null);
+
+  useEffect(() => {
+    setThemeColors(getRandomAreaPlotAnimationThemeColors());
+  }, [activeThemes]);
```

Comment:
Most of the places that depend on resolving css vars need to go in an effect so that the latest data is available to the DOM right after a theme selection change. 

I'm not 100% sure, but I believe this happens because ThemeProvider renders style tags dynamically which would be in the same render as when useMemo gets calculated vs useEffect would fire after the DOM has actually updated the colors that are needed for the resolve.

For this particular example, it's easiest to see in the styleguide. When I tried useMemo and change the theme, the change trails behind till the next render

Reply:
I added a clarifying comment to the effect

Reply:
Okay - as we discussed in our meeting it would be nice if we pulled this logic up to the `ThemeProvider` so we don't need to do this pattern in so many places (here, `ChartThemeProvider`, `IrisGridThemeProvider`, etc).
So in `ThemeProvider`, the `value` it provides to the `ThemeContext.Provider` is only set _after_ a `useLayoutEffect` occurs in the `ThemeProvider`. Then any child can just do `useActiveThemes` (or maybe a `useThemeColors` type of hook?) and just get the active colours without needed to add their own `useLayoutEffect`
Can be a follow up ticket.